prov/usnic/src/usdf_cq.c

```diff
@@ -496,12 +475,13 @@ usdf_cq_read_common_soft(struct fid_cq *fcq, void *buf, size_t count,
 	last = entry + (entry_len * count);
 	tail = cq->c.soft.cq_tail;
 
-	// XXX ... handle error comps
 	while (entry < last && tail != cq->c.soft.cq_head) {
+		if (tail->cse_prov_errno > 0) {
+			return -FI_EAVAIL;
+		}
 		memcpy(entry, tail, entry_len);
```

Comment:
Prankur and I chatted about this offline.  It technically worked correctly because of careful element selection/layout for `usdf_cq_soft_entry`, but he's going to stick a new `static inline` helper in place to be more robust and clearly communicate intent here.